R/ensure-tpch-source.R

```diff
@@ -4,6 +4,16 @@
 #' @export
 tpch_tables <- c("customer", "lineitem", "nation", "orders", "part", "partsupp", "region", "supplier")
 
+#' Generate tpch data
+#'
+#' Generate tpch data at a given scale factor. By default,
+#' `data` is assumed to be relative to the current working directory, but
+#' you can set the environment variable `ARROWBENCH_DATA_DIR` to point to another
+#' (permanent) base directory.
+#'
+#' @param scale_factor a relative measure of the size of data in gigabytes.
+#'
+#' @export
 generate_tpch <- function(scale_factor = 1) {
```

Comment:
Here's a branch with this via datalogistik: https://github.com/alistaire47/arrowbench/blob/feat/datalogistik/R/ensure-tpch-source.R If we're using this directly, we could just use datalogistik directly instead, though there is a nice util for parsing the json, so maybe this is nicer still.